resource-manager/src/main/java/org/snomed/module/storage/ModuleStorageCoordinator.java

```diff
@@ -0,0 +1,602 @@
+     *
+     * @param codeSystem    CodeSystem of RF2 package.
+     * @param moduleId      Most important, or identifying, module id of RF2 package.
+     * @param effectiveTime Effective time of RF2 package.
+     * @return Boolean indicating success of operation.
+     */
+    public boolean archive(String codeSystem, String moduleId, String effectiveTime) {
+        if (!allowArchive || invalidArguments(codeSystem, moduleId, effectiveTime)) {
+            return false;
+        }
+
+        try {
+            return doArchiveRelease(codeSystem, moduleId, effectiveTime);
+        } catch (Exception e) {
+            LOGGER.error("Cannot archive; exception encountered", e);
+            return false;
+        }
+    }
+
+    /**
+     * Download ModuleMetadata stored for given arguments, as well as optionally downloading dependent ModuleMetadata.
+     *
+     * @param codeSystem          CodeSystem of RF2 package.
+     * @param moduleId            Most important, or identifying, module id of RF2 package.
+     * @param effectiveTime       Effective time of RF2 package.
+     * @param includeFile         Whether to download RF2 package.
+     * @param includeDependencies Whether to download dependent ModuleMetadata.
+     * @return Collection of dependent ModuleMetadata, sorted by effective time.
+     */
+    public List<ModuleMetadata> downloadRelease(String codeSystem, String moduleId, String effectiveTime, boolean includeFile, boolean includeDependencies) {
+        if (invalidArguments(codeSystem, moduleId, effectiveTime)) {
+            return Collections.emptyList();
+        }
+
+        return doDownloadRelease(codeSystem, moduleId, effectiveTime, includeFile, includeDependencies);
+    }
+
+    /**
+     * Download ModuleMetadata stored for given arguments, as well as downloading dependent ModuleMetadata.
+     *
+     * @param codeSystem    CodeSystem of RF2 package.
+     * @param moduleId      Most important, or identifying, module id of RF2 package.
+     * @param effectiveTime Effective time of RF2 package.
+     * @return Collection of dependent ModuleMetadata, sorted by effective time.
+     */
+    public List<ModuleMetadata> downloadRelease(String codeSystem, String moduleId, String effectiveTime) {
+        if (invalidArguments(codeSystem, moduleId, effectiveTime)) {
+            return Collections.emptyList();
+        }
+
+        return doDownloadRelease(codeSystem, moduleId, effectiveTime, true, true);
+    }
+
+    /**
+     * Update ModuleMetadata stored for given arguments.
+     *
+     * @param codeSystem    CodeSystem of RF2 package.
+     * @param moduleId      Most important, or identifying, module id of RF2 package.
+     * @param effectiveTime Effective time of RF2 package.
+     * @return Boolean indicating success of operation.
+     */
+    public boolean setPublished(String codeSystem, String moduleId, String effectiveTime, boolean published) {
```

Comment:
@pgwilliams by default, the `published` and `edition` properties of `ModuleMetadata` will be false. The `setPublished` and `setEdition` methods can be used by the client/caller to update them accordingly.